selfhost/main.jakt

```diff
@@ -472,65 +573,166 @@ function main(args: [String]) {
-        } else {
-            let basename_without_extension = file_path.basename().substring(start: 0, length: file_path.basename().length() - 5)
-            cpp_filename = binary_dir + "/" + basename_without_extension + ".cpp"
-            output_filename = binary_dir + "/" + basename_without_extension
+    if not File::exists(binary_dir) {
+        make_directory(path: binary_dir)
+    }
+
+    for (file, contents_and_path) in codegen_result.iterator() {
+        let (contents, module_file_path) = contents_and_path
+
+        let path = binary_dir + "/" + file
+        try write_to_file(data: contents, output_filename: path) catch error {
+            eprintln("Error: Could not write to file: {} ({})", file, error)
+            return 1
+        }
+
+        if generate_depfile.has_value() and file.ends_with(".cpp") {
+            let escaped = file.replace(replace: " ", with: "\\ ")
+            let escaped_module_file_path = module_file_path.replace(replace: " ", with: "\\ ")
+            depfile_builder.append_string(format(
+                "{} {}.h: {}"
+                escaped
+                escaped.substring(start: 0, length: file.length() - 4)
+                escaped_module_file_path
+            ))
+            depfile_builder.append(b'\n')
         }
+    }
 
+    if generate_depfile.has_value() {
         try {
-            write_to_file(data: output, output_filename: cpp_filename)
+            write_to_file(
+                data: depfile_builder.to_string()
+                output_filename: generate_depfile!
+            )
         } catch error {
-            eprintln("Could not write file: {} ({})", cpp_filename, error);
-            return error.code();
+            eprintln("Error: Could not write to file list ({})", error)
+            return 1
         }
+    }
+
+    if build_executable or run_executable {
+        mut ids: {usize} = {}
+        mut linked_files: [String] = []
+        mut pool = ParallelExecutionPool::create(max_concurrent)
+
+        mut total_jobs = 0
 
-        if prettify_cpp_source {
-            mut command = clang_format_path + " -i " + cpp_filename
-            if dot_clang_format_path.has_value() {
-                command+= " --style=file:" + dot_clang_format_path!
+        for (file_name, _) in codegen_result.iterator() {
+            if file_name.ends_with(".h") {
+                continue
             }
-            system(command.c_string())
-        }
-    } else {
-        println("{}", output)
-    }
-
-    if (build_executable or run_executable) {
-        let compiler_status = run_compiler(
-            cxx_compiler_path
-            cpp_filename
-            output_filename
-            runtime_path
-            extra_include_paths
-            extra_lib_paths
-            extra_link_libs
-            optimize
-        )
+            total_jobs += 1
+        }
 
-        if run_executable and compiler_status == 0 {
-            return system(output_filename.c_string())
```

Comment:
Seems that currently the option to run the compiled executable has been dropped.

Reply:
Oh _that's_ the commented stuff I deleted and forgot about!